### PR TITLE
STORM_B: Create worker page in admin panel

### PR DESCRIPTION
**Pull request by AndriiBuhalo**
image
image
image
image
image

1. Add new page worker in admin panel. 
2. Create CRUD with use turbo frame 
3.  Add locale for all page
4. Rspec test

### CODE REVIEW

File: app/views/dredd/workers/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            <%= t("dredd.users.actions") %>
+          </p>
+        </div>
+        <%= turbo_frame_tag "new-dredd-worker-form" , src: new_dredd_worker_path %>
+          <div id="workers" class="border-b">
+            <%= render @workers %>
+          </div>
+    </div>
+  </div>
+  <%= pagy_nav(@pagy).html_safe if @pagy.pages> 1 %>
```
Comment (Shabaldas) [inline app/views/dredd/workers/index.html.erb:39]: same

File: spec/request/dredd/workers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    it 'create a new worker' do
+      expect do
+        post dredd_workers_path, params: {
+          worker: {
+            first_name: 'Vasa',
+            last_name: 'Pupkin',
+            nickname: 'Vasa 111'
+          }
+        }
+      end.to change(Worker, :count).by(1)
```
Comment (Shabaldas) [inline spec/request/dredd/workers_spec.rb:42]:
```ruby
as: :turbo_stream
```

Should be present in all CRUD actions

File: spec/request/dredd/workers_spec.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    it 'update worker' do
+      expect do
+        put(dredd_worker_path(worker), params:)
+        worker.reload
+      end.to change(Worker, :count).by(0) # rubocop:disable RSpec/ChangeByZero
+          .and change(worker, :last_name).from(worker.last_name).to('Pupkin')
+          .and change(worker, :nickname).from(worker.nickname).to('Vasa 111')
+      worker = Worker.last
+      expect(worker.last_name).to eq('Pupkin')
```
Comment (Shabaldas) [inline spec/request/dredd/workers_spec.rb:83]: extra rspec, you already check this

File: spec/request/dredd/workers_spec.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      expect(worker.nickname).to eq('Vasa 111')
```
Comment (Shabaldas) [inline spec/request/dredd/workers_spec.rb:84]: same

File: spec/request/dredd/workers_spec.rb [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      expect(response.body).to include('Worker was successfully updated.')
+    end
+  end
+
+  describe 'DELETE /dredd/workers/:id' do
+    let(:worker) { create(:worker, user_id: user.id) }
```
Comment (Shabaldas) [inline spec/request/dredd/workers_spec.rb:90]: do you nend `user_id` here?

File: app/views/dredd/workers/_worker.html.erb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            true, form_class: "inline-block" do %>
+              <svg class="inline-block w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24"
+                xmlns="http://www.w3.org/2000/svg">
+                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
+                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16">
+                </path>
+              </svg>
+            <% end %>
+              <%= link_to edit_dredd_worker_path(worker) do %>
+                <svg class="inline-block w-6 h-6 ml-6" fill="none" stroke="currentColor" viewBox="0 0 24 24"
```
Comment (Shabaldas) [inline app/views/dredd/workers/_worker.html.erb:30]: move icons to svg folder and user image_tag here

File: app/views/dredd/workers/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<%= form_with model: [:dredd, worker], id: dom_id(worker, "form" ), class: "" do |form| %>
```
Comment (Shabaldas) [inline app/views/dredd/workers/_form.html.erb:1]: if you don't use `class: ""` you can remove it.

File: app/controllers/dredd/workers_controller.rb [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    end
+
+    def destroy
+      @worker.destroy
+      flash.now[:notice] = 'Worker was successfully destroyed.'
+      render turbo_stream: [
+        turbo_stream.remove(@worker),
+        turbo_stream.update('notice', partial: 'layouts/flash')
+      ]
+    end
```
Comment (Shabaldas) [inline app/controllers/dredd/workers_controller.rb:54]: try this one, and if it's works. did this logic in other actions.
```ruby

def destroy
      @worker.destroy
      flash.now[:info] = { text: 'Workerwas successfully destroyed', icon: 'success_icon' }.stringify_keys
      render turbo_stream: [
        turbo_stream.remove(@worker),
        turbo_stream.prepend('flash', partial: 'dredd/shared/flash')
      ]
    end
```

File: app/controllers/dredd/workers_controller.rb [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+      if @worker.update(worker_params)
+        flash.now[:notice] = 'Worker was successfully updated.'
+        render turbo_stream: [
+          turbo_stream.replace(@worker, @worker),
+          turbo_stream.update('notice', partial: 'layouts/flash')
+        ]
+      else
+        render :edit, status: :unprocessable_entity
+      end
+    end
```
Comment (Shabaldas) [inline app/controllers/dredd/workers_controller.rb:45]: tyr this one
```ruby
def update
      if @worker.update(worker_params)
        flash.now[:notice] = { text: 'Workerwas successfully updated', icon: 'success_icon' }.stringify_keys
        render turbo_stream: [
          turbo_stream.replace(@worker, @worker),
          turbo_stream.prepend('flash', partial: 'dredd/shared/flash')
        ]
      else
        render :edit, status: :unprocessable_entity
      end
    end
```

File: app/controllers/dredd/workers_controller.rb (hunk 8 above)
Comment (AndriiBuhalo) [reply app/controllers/dredd/workers_controller.rb:53]: done)

File: app/controllers/dredd/workers_controller.rb (hunk 9 above)
Comment (AndriiBuhalo) [reply app/controllers/dredd/workers_controller.rb:44]: done)

File: config/credentials.yml.enc [hunk 10]
```diff
@@ -1 +1 @@
...
...KnttBWA+bcwnyPL4wQLtcrankoDzYJH6gX9mk6Fw9/kp5396Ye+ztgLYb1IvUyan2iTiiGNF+Mjz7soLZoX2xl9pMf65GIMIRvDOSyIrLdTb1DyUNIHDMVQSkCi6QpRS6eM5mZ/V9MJKhABXhZC98edCGgvPmaaaXmHJvpOPtWXQz6ukp7eJCZlQ8uGA0by02Yr409VdV7vyukwoq4IWcQg3mQmLrvF8a+KLxwyHvqDIJ5Sok/ZZ29LUzfWMBMUN+IAp1uEv+BjbyqfwTb/IiKRijaQIfnSaEXyxwbhyVugiW+RrIIGaM5v3hoX3Dy75lS51aa7ElxTnXaoKcjg5aFL1a66EZIlr4tJ8OJv4RhOHTn99YhgX99fXvYE64vr5o0C0Dgy8ABWuFZpsIb8IIkQc499DAGhwAIbyB18stTqHXou3heuKq2tMKKKYprx6uioMJGV3yjqQ3o+YOP9t+u0z5515CPAv9Xf52QrQrOGwNY2Ro5BLESCPkUtsc204BiQnndEzIsoXLpRvJ7Ar2lZhZXZHs8LSLKDNaTJyMGrs1XDEnfp1oHugROiyf2PXH80DjUjk7r6p2UYDI0/t6QqieRi8a5fR6BE8pUG1Oh6XEAXRBAMk1vJ3XjFiP7ZN8y8rFxJLkajWLg1HbBOa8BoGSk2P5pUScSEM+qzIQ7pbPCzB/3fODTB0M375LeTfRZSTJRGHc0lnckrKC38y1oHXUPXenFSfx7iO0slIOLH3RRKACioasGqzv5nYVAu+KLnwvTw/elXSkxbN6O7lfvnedKV8OLHbTnPfvNIkiaPOHRlB2WDtacQzwJfasZZPmTifoF90FljjtI3qRpku+B6vg9dgYQvVhWCPSNtwma57RmLD3JBMEmv4XN0TGwh7JAIaUjY8WbCvEsX7FNCjDUb099gSooQrJJG2NeL207Ca7EKCQzKAHA9PVT1l6E8EzlJ8v3Cz/e2SwIjthTCSqt10Fn6P1U/mDB5lJ8YnRebYzDEh3TU6tdlwdntuXVkA6H7xMe3en9DIJB6gmtO/3Ohxn2M4YmJUG/qn05faXABpcIHzG+CxwVHE66woVDHXkh6gUO2hP7VpAHN0sn6/yfT3GcmyLFHnnzg96jDrUF89fcVVU6plVsKh1tGWlhos4RA0--sZjR76Dl949WO4ur--637Au4qOlTIZ0hfHFcqHhg==
```
[the remark this answers is not in the document]
Comment (AndriiBuhalo) [reply config/credentials.yml.enc:1]: done)

File: spec/request/dredd/workers_spec.rb (hunk 2 above)
Comment (AndriiBuhalo) [reply spec/request/dredd/workers_spec.rb:42]: done)

**Review by Shabaldas — APPROVED** [review]

**Pull request merged**